### DISCUSSION TITLE
Update AIO to 3.2 (NC25) failed

### DISCUSSION THREAD

**Discussion by Michael-Muthmann** [nextcloud/all-in-one#1609] [Questions] [answered]
I installed Nextcloud AIO 2 month ago, with no Problems. During Setup i've choosen to use an external mysql server on another host. The first update went flawlessly. Today i tried to update to the latest version. 
Now the nextcloud-aio-nextcloud container is unable to start. 
This is the log:

Waiting for database to start...
nc: getaddrinfo: Name does not resolve
              now              
-------------------------------
 2022-12-27 15:14:40.452932+01
(1 row)
Installing imagemagick via apk...
Installing PHP build dependencies...
Installing Imagick via PECL...
Configuring Redis as session handler...
Setting php max children...
An unhandled exception has been thrown:
Doctrine\DBAL\Exception: Failed to connect to the database: An exception occurred in the driver: SQLSTATE[HY000] [1045] Access denied for user 'oc_nextcloud'@'192.168.0.175' (using password: YES) in /var/www/html/lib/private/DB/Connection.php:139
.......
#38 /var/www/html/cron.php(43): require_once('/var/www/html/l...')
#39 {main}
Error: 
× Failed to connect to database
  ├─▶ error returned from database: 1045 (28000): Access denied for user
  │   'oc_nextcloud'@'192.168.0.175' (using password: YES)
   ╰─▶ 1045 (28000): Access denied for user
     'oc_nextcloud'@'192.168.0.175' (using password: YES)

The dbuser, name and host  i previously configured are ignored, even the config file was changed.

**previously configured:**
 'dbtype' => 'mysql',   
'dbname' => 'nextcloud',
'dbhost' => '192.168.0.163:3306',
'dbtableprefix' => 'oc_',
'dbuser' => 'oc_mimu_Admin',
'dbpassword' => 'xxxxxxxx',
'installed' => true,

**now the config looks like this:**
 'dbtype' => 'mysql',                                             
  'version' => '24.0.7.1',                          
  'dbname' => 'nextcloud',                                       
  'dbhost' => '192.168.0.163',                                       
  'dbport' => '3306',                                            
  'dbtableprefix' => 'oc_',                                          
  'dbuser' => 'oc_nextcloud',                                        
  'dbpassword' => 'xxxxxxxxx',
  'installed' => true,   

Db user and password are changed. why is it trying to connect to the ip of the dockerhost?
I already tried to change the dbuser and password in the config file, but it still tries to connect the wrong dbserver.

Unfortunately i dont' have a backup.
I am thankful for all help and information.

**Comment by szaimen** [comment, marked as answer]
Hi, I am honest to you: I don't have much motivation to help if you do not create backups.

Also, we state pretty clearly that you need to know what you do when using the manual-install which you've shown to not being capable of. However I might agree that we did not state anywhere that mysql is not a supported database. However latest since v4.0.1 is this the case since the pdo_mysql driver is not included by default anymore.

---

As for you problem probably two things are needed:
1. add `pdo_mysql` to this variable: https://github.com/nextcloud/all-in-one/blob/209d2e0f9a3138854d8bd6b54ee56f2b266ed3f9/manual-install/sample.conf#L16
2. Modify https://github.com/nextcloud/all-in-one/blob/209d2e0f9a3138854d8bd6b54ee56f2b266ed3f9/manual-install/latest.yml#L64 to `mimu_Admin`

However there is no guarantee that things will work afterwards since you cannot restore a backup from before the breakage.
